### PR TITLE
Rename `AxArchVCpuExitReason` to `AxVCpuExitReason` and add `VMCall` to it

### PR DESCRIPTION
**Pull request by hky1999**

### CODE REVIEW

File: src/exit.rs [hunk 1]
```diff
@@ -45,8 +46,18 @@ impl From<AccessWidth> for usize {
 type Port = u16;
 
 /// The result of [`AxArchVCpu::run`].
+/// Can we reference or directly reuse content from [kvm-ioctls](https://github.com/rust-vmm/kvm-ioctls/blob/main/src/ioctls/vcpu.rs) ?
 #[non_exhaustive]
-pub enum AxArchVCpuExitReason {
+#[derive(Debug)]
+pub enum AxVCpuExitReason {
+    VMCall {
```
Comment (aarkegz) [inline src/exit.rs]: Add a doc comment here.

File: src/exit.rs [hunk 2]
```diff
@@ -103,4 +115,9 @@ pub enum AxArchVCpuExitReason {
     ///
     /// This exists to allow the caller to have a chance to check virtual devices/physical devices/virtual interrupts.
     Nothing,
+    /// Corresponds to KVM_EXIT_FAIL_ENTRY.
+    FailEntry(
```
Comment (aarkegz) [inline src/exit.rs]: Maybe we should make fields of this variant named?

File: src/exit.rs [hunk 3]
```diff
@@ -45,8 +46,19 @@ impl From<AccessWidth> for usize {
...
+#[derive(Debug)]
+pub enum AxVCpuExitReason {
+    /// The instruction executed by the vcpu performs a hypercall.
+    Hypercall {
+        /// The hypercall number.
+        nr: u64,
+        /// The arguments for the hypercall.
+        args: [u64; 6],
+        /// Whether the hypercall was executed in long mode.
+        longmode: u32,
```
Comment (aarkegz) [inline src/exit.rs]: Is this field necessary? If so, can we make it a `bool`?

Comment (hky1999) [reply src/exit.rs]: Actually, both `HyperCall` and `FailEntry` was copied from [kvm-ioctls](https://github.com/rust-vmm/kvm-ioctls/blob/main/src/ioctls/vcpu.rs)

Comment (aarkegz) [reply src/exit.rs]: Copying without rustifying seems less convincing🙂

Comment (hky1999) [reply src/exit.rs]: I‘ll just delete the redundant useless fields & parametera for now.

**Review by aarkegz — APPROVED** [review]

**Pull request merged**